tests/legacy-cli/e2e/utils/process.ts

```diff
@@ -33,11 +33,22 @@ function _exec(options: ExecOptions, cmd: string, args: string[]): Promise<Proce
   let stdout = '';
   let stderr = '';
   const cwd = options.cwd ?? process.cwd();
-  const env = options.env;
+  const env = options.env ?? process.env;
   console.log(
     `==========================================================================================`,
   );
 
+  const paths = (env.PATH || process.env.PATH)!.split(delimiter);
```

Comment:
This seems redundant.

You are splitting here for the sake of having an array, but you are not doing anything with this. Also, `unshift` IMHO doesn't aid readability when used multiple times as you are sort of fighting the flow. (Do you get what I mean)?

I feel like something like the below is more readable and avoids unnecessary split.

```ts
const paths = [
  join(getGlobalVariable('npm-global'), process.platform.startsWith('win') ? 'bin' : ''),
  join(getGlobalVariable('yarn-global'), 'bin'),
  env.PATH || process.env['PATH'],
].join(delimiter);
```

Also, I think that `yarn_global` should take precedence over `npm_global`.

Reply:
I agree that is clearer. Why yarn over npm? (I've put yarn first for now, just curious...)

Reply:
If you use Yarn, I feel like it is expected to prefer to resolve the global binary from the Yarn bin before fallbacking to the NPM bin.